src/comm.cuh

```diff
@@ -99,31 +95,27 @@ send_data_by_offset(
         // calculate the starting address
         const void *start_addr = (void *)((char *)data + offset[itarget_rank] * item_size);
 
-        // send buffer to the target node
-        requests[itarget_rank] = communicator->send(
-            start_addr, count, item_size, itarget_rank, offset_tag
-        );
+        // send buffer to the target rank
+        communicator->send(start_addr, count, item_size, itarget_rank);
     }
-
-    return requests;
 }
 
 
 /**
  * Receive the data sent from 'send_data_by_offset'.
  *
+ * Note: This call should be enclosed by communicator->start() and communicator->stop().
+ *
  * @param[out] data         The data received from each rank. This argument does not need to be preallocated, but the
  *                          caller is responsible for freeing this buffer using RMM_FREE. See
```

Comment:
Are we still using `RMM_FREE`? Should this get updated?

Reply:
Nice catch! Updated.